File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/template/OrganizationTemplateAssembler.java
package de.helfenkannjeder.helfomat.api.template;

import de.helfenkannjeder.helfomat.core.template.GroupTemplate;
import de.helfenkannjeder.helfomat.core.template.OrganizationTemplate;

import java.util.List;
import java.util.stream.Collectors;

public class OrganizationTemplateAssembler {

    public static OrganizationTemplateDto toOrganizationTemplateDto(OrganizationTemplate organizationTemplate) {
        return new OrganizationTemplateDto(
            organizationTemplate.getName(),
            toGroupTemplates(organizationTemplate.getGroups())
        );
    }

    private static List<GroupTemplateDto> toGroupTemplates(List<GroupTemplate> groups) {
        return groups.stream()
            .map(OrganizationTemplateAssembler::toGroupTemplate)
            .collect(Collectors.toList());
    }

    private static GroupTemplateDto toGroupTemplate(GroupTemplate groupTemplate) {
        return new GroupTemplateDto(
            groupTemplate.getName(),
            groupTemplate.getDescription()
        );
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/template/OrganizationTemplateAssembler.kt
package de.helfenkannjeder.helfomat.api.template

import de.helfenkannjeder.helfomat.core.template.GroupTemplate
import de.helfenkannjeder.helfomat.core.template.OrganizationTemplate

fun OrganizationTemplate.toOrganizationTemplateDto() = OrganizationTemplateDto(this.name, this.logoSuggestions, this.groups.toGroupTemplates())

fun List<GroupTemplate>.toGroupTemplates() = this.map { it.toGroupTemplate() }
fun GroupTemplate.toGroupTemplate(): GroupTemplateDto = GroupTemplateDto(this.name, this.description)